node/src/components/contract_runtime/config.rs

```diff
@@ -21,6 +22,10 @@ pub struct Config {
     ///
     /// Defaults to 512.
     max_readers: Option<u32>,
+    /// The limit of depth of recursive global state queries.
+    ///
+    /// Defaults to 5.
+    query_depth_limit: Option<u64>,
```

Comment:
I'd be inclined to name it `max_query_depth` for consistency with the other limiting config values.